src/Polly.Core/Strategy/OutcomeGenerator.cs

```diff
@@ -3,26 +3,16 @@
 
 namespace Polly.Strategy;
 
-#pragma warning disable S2436 // Types and methods should not have too many generic parameters
-#pragma warning disable CA1005 // Too many generic arguments, but this is the only way to make the API work without unnecessary extensions
-
 /// <summary>
 /// The base class for all generators that generate a value based on the <see cref="Outcome{TResult}"/>.
 /// </summary>
-/// <typeparam name="TGeneratedValue">The type of generated value.</typeparam>
 /// <typeparam name="TArgs">The arguments the generator uses.</typeparam>
-/// <typeparam name="TSelf">The class that implements <see cref="OutcomeGenerator{TGeneratedValue, TArgs, TSelf}"/>.</typeparam>
-public abstract partial class OutcomeGenerator<TGeneratedValue, TArgs, TSelf>
+/// <typeparam name="TGeneratedValue">The type of generated value.</typeparam>
+public sealed partial class OutcomeGenerator<TArgs, TGeneratedValue>
```

Comment:
Maybe `TResult` or `TOutcome` instead of `TGeneratedValue`?

Reply:
`TResult` collides with generic methods that have `TResult` and I feel `TOutcome` collides with the `Outcome` class we already have. The generated value is not really an outcome.

Reply:
The naming just feels a bit weird to me.

The type is described as a _"generator of an outcome"_, so it feels to me that its output would be an _"outcome"_, not a _"generated"_. A bit like a _"foo factory"_ produces _"foo-s"_ rather than _"fabricated-s"_.

What if we drop the `Geneated` prefix, and just have `TValue`? Or will that clash elsewhere too?

Reply:
Nope, `TValue` is fine. We can use it.